pkg/middleware/middleware_test.go

```diff
@@ -571,3 +573,88 @@ func middlewareScenario(t *testing.T, desc string, fn scenarioFunc) {
 		fn(sc)
 	})
 }
+
+func TestDontRotateTokensOnCancelledRequests(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	reqContext, _ := initTokenRotationTest(ctx)
```

Comment:
I'd prefer if we don't ignore the error here. I can update the branch with an error check?

Reply:
Never mind, it's not an error :)